src/Symfony/Bridge/Twig/Extension/AssetMapperRuntime.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\AssetMapper\AssetMapperInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class AssetMapperRuntime
+{
+    public function __construct(
+        private readonly AssetMapperInterface $assetMapper,
+    ) {
+    }
+
+    public function source(string $logicalPath): string
+    {
+        if (!$asset = $this->assetMapper->getAsset($logicalPath)) {
+            throw new \LogicException(\sprintf('The asset "%s" was not found.', $logicalPath));
+        }
+
+        if ($asset->content) {
+            return $asset->content;
+        }
+
+        if (false === $content = file_get_contents($asset->sourcePath)) {
+            throw new \RuntimeException(\sprintf('The asset "%s" could not be read.', $asset->sourcePath));
+        }
```

Comment:
I'm not sure here, or maybe the name is to reconsider. 

`$asset->content`, when provided, will always be different from `file_get_contents($asset->sourcePath)`

```php
    /**
     * The final content of this asset if different from the sourcePath.
     *
     * If null, the content should be read from the sourcePath.
     */
    public readonly ?string $content;
```

So i'm wondering if you should not always return the original content here 🤷‍♂️

Reply:
Perhaps a different name than "source". This logic is definitely important for this feature (to be able to handle sass/tailwind/etc).

Reply:
I may have missunderstood... do you want the source code of the file... or the compiled one ?

Reply:
Compiled if available or the source of not.

My example in the PR description: you can imagine foundation CSS being just the source and style.css being complied via tailwind.

Reply:
This was what made me question the name.. So in the end this is the content of the mapped asset more than its source code.

Maybe `mapped_asset_content` ? `asset_mapper_compiled_content` ?

Reply:
I see, you're right, content makes more sense.